File: JavaScriptSolutions/005_smallestMult.js
const isPrime = num => {
    if (num <= 1) return false;
    if (num === 2 || num === 3) return true;
    if (num % 2 === 0) return false;
        
    for (let i = 3; i <= Math.sqrt(num); i += 2) {
        if(num % i === 0) return false;
    }

    return true;
}

const smallestMult = num => {
    let smallest = 1;
    for (let i = 2; i <= num; i++) {
        if (isPrime(i)) {
            let factor = i;
            while (factor * i <= num) {
                factor *= i;
            }
            smallest *= factor;
        }        
    }
    return smallest;
}

File: PythonSolutions/005_smallestMult.py
import math

def isPrime(num):
    if num <= 1 : return False
    if num == 2 or num == 3: return True
    if num % 2 == 0 or num % 3 == 0: return False
    for i in range(3, round(math.sqrt(num)) + 1, 2):
        if num % i == 0: return False
    return True

def smallestMult(num):
    smallest = 1
    for i in range(2, num + 1):
        if isPrime(i):
            factor = i
            while factor * i <= num: factor = factor * i
            smallest = smallest * factor
    return smallest